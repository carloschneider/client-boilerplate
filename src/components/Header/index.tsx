import * as S from './styles'

export type HeaderProps = {
  content: string
}

const Header = ({ content }: HeaderProps) => (
  <S.Wrapper>
    <h1>{content}</h1>
  </S.Wrapper>
)

export default Header
