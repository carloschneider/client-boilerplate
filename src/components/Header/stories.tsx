import { Story, Meta } from '@storybook/react/types-6-0'

import Header, { HeaderProps } from '.'

export default {
  title: 'Header',
  component: Header,
} as Meta

const Template: Story<HeaderProps> = (args) => <Header {...args} />

export const Default = Template.bind({})
Default.args = {
  content: 'Header',
}
