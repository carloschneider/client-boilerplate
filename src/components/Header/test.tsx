import { render, screen } from '@testing-library/react'

import Header from '.'

describe('<Header />', () => {
  it('should render the component', () => {
    const { container } = render(<Header content="Header" />)
    const heading = screen.getByRole('heading', { name: /Header/i })

    expect(heading).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
