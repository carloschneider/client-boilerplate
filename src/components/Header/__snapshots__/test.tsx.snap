// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the component 1`] = `
.c0 {
  text-align: center;
}

<div
  class="c0"
>
  <h1>
    Header
  </h1>
</div>
`
